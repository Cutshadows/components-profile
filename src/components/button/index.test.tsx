import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ButtonSite from './index';
import jest from 'jest-mock';

describe('Button', () => {
	it('renders without crashing', () => {
		render(<ButtonSite children={undefined} attributes={undefined} />);
		expect(screen.getByRole('button')).toBeInTheDocument();
	});

	it('renders the correct text', () => {
		const attributes = { onClick: () =>{}, name:'btn-name', text: 'Click me', className:"btn"};
		const {container} = render(<ButtonSite children={undefined} attributes={attributes} />);
		expect(container.querySelector('btn')).toMatchSnapshot();
	});

	it('calls the onClick function when clicked', () => {
		const onClickMock = jest.fn(); //ReferenceError: jest is not defined 
		const attributes = { onClick: onClickMock, name:'btn-name', text: 'Click me', className:"btn"};
		const { getByRole } = render(<ButtonSite attributes={attributes} children={undefined} />);
		const buttonElement = getByRole('button');
		fireEvent.click(buttonElement);
		expect(onClickMock).toHaveBeenCalled();
	});
});