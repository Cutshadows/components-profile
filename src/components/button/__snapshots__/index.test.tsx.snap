// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders the correct text 1`] = `null`;
